Woof/Woof/View/OwnerProfileView/OwnerProfileView.swift

```diff
@@ -0,0 +1,55 @@
+import SwiftUI
+
+/// View which could be used to display or edit information about Owner
+struct OwnerProfileView: View {
+    /// View model responsible to manage data from model layer
+    @ObservedObject var viewModel = OwnerProfileViewModel()
+
+    /// Owner information editing status tracker
+    @State private var isEditingTracker = false
+
+    var body: some View {
+        VStack {
+            if isEditingTracker {
+                EditOwnerInformationView(name: $viewModel.name,
+                                         surname: $viewModel.surname,
+                                         phone: $viewModel.phone,
+                                         address: $viewModel.address)
+            } else {
+                OwnerCardView(name: viewModel.name,
+                              surname: viewModel.surname,
+                              phone: viewModel.phone,
+                              address: viewModel.address,
+                              avatarUrl: viewModel.avatarURL)
+            }
+            Spacer()
+        }
+        .padding(.horizontal)
+
+        .overlay(alignment: .topTrailing) {
+            Button(action: {
+                if !viewModel.name.isEmpty {
+                    viewModel.save()
+                    isEditingTracker.toggle()
+                }
```

Comment:
I don't understand the logic. Why do you verify the name is not empty?
For now, it always will be true, because we have a default value `User` for a name when initializing the model, and we restrict it as a mandatory field.
Plus you trigger the save method as well when we switch to edit mode.
```suggestion
                if isEditingTracker {
                    viewModel.save()
                }
                isEditingTracker.toggle()
```

Reply:
But we never save information if `viewModel.save()` method is called when name is empty

Reply:
I thought from the point of view that we always save (update) information when the name is not empty, so I wrote a code that triggers this.

Reply:
I use the option you suggested, it triggers the save method by the value of the `isEditingMode` var, it looks more logical this way.